### DISCUSSION TITLE
【打卡】重复的重构

### DISCUSSION THREAD

**Discussion by swlws** [swustzzh/coder_books#15] [General]
书名：《重构 改善既有代码的设计第二版》

章节：第一章

什么是好代码？

好代码的检验标准就是人们是否能轻而易举地修改它，有人需要修改代码时，他们应能轻易找到修改点，应该能快速做出更改，而不易引入其他错误

**技巧：**

一个 for 循环，和两个同级的 for 循环，大多数情况下，性能消耗类似，差距可以忽略不计。

**思考：**

1. 开始重构后，每一步修改后，都要记得跑一次测试用例，确定每一步的修改都是正确的。
   - 即小步修改，每次修改后就运行测试。
   - 避免修改太多，陷入测试的麻烦。
2. 变量命名。
   - 语义化
   - 优美，避免太长
   - 不能太通用，比如：format、make、readFile。尽量与业务、功能有关联性
3. 如果重构引入了性能损耗，先完成重构，再做性能优化。

**Comment by swustzzh** [comment]
感觉重构之前还是要先确定规范，根据规范修改逻辑

**Reply by swlws** [reply]
这里的规范应该是编码规范，每个组织都有自己的编码规范，比如[阿里巴巴 Java 开发规范](https://developer.aliyun.com/article/1170598)

重构，本质上是对设计模式的应用，通用设计模式变更代码结构，更易扩展、更易维护
